test/sql/types/timestamp/test_timestamp_to_tz_cast.test

```diff
@@ -0,0 +1,245 @@
+# name: test/sql/types/timestamp/test_timestamp_to_tz_cast.test
+# description: Test casting to TIMESTAMP WITH TIME ZONE from any unit
+# group: [timestamp]
+
+require icu
+
+# This breaks the 'verification_stats' check
```

Comment:
Should this not be fixed before merging?

Reply:
Ideally yes, I couldn't make sense of it though
It seems like there is a bug in the verification code, which has no impact on the correctness of the database? But I could be wrong on that

Reply:
The `verification_stats` checks that stats are propagated correctly in the statistics propagation. If the stats are wrong there could be incorrect results leading from e.g. follow-up filters based on the columns with wrong stats.

Reply:
That makes sense, but the name `verification_stats`, and how it's not passed by default to CastParameters makes me think this is completely unused besides verification.
Which leads me to believe that the actual stats are being recorded correctly at a later point.

Reply:
They're also part of the Expression, not the result Vector which feels like a disconnect to me.
I'm now looking into providing them to the CastParameters, only to find the `verification_stats` are nullptr at this point.

Reply:
> They're also part of the Expression, not the result Vector which feels like a disconnect to me.

Statistics are set for the entire result of the expression, and e.g. the min/max means that no value can ever come out of this expression that is higher/lower than what the statistics specify

Reply:
I'm happy to fix this, but I can't really seem to find where this needs to be fixed.
If the verifications stats are only set when the flag is enabled, then I don't think that just passing along the verification stats to the cast function is enough?

That sounds like it's only fixing the part we verify, not touching the actual stats gathering.
That's my original issue with this, they feel disjointed - the actual stats and the verification stats

Reply:
Laurens pointed me in the right direction, this was missing cases in `propagate_cast.cpp`. I've nullified the stats for now when casting between different timestamp units